src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -109,26 +121,21 @@ def to_interpreter_version(self) -> Optional[Tuple[int, int]]:
         return int(mo.group("major")), int(mo.group("minor"))
 
 
-class PythonAWSLambda(Target):
-    alias = "python_awslambda"
-    core_fields = (
+class PythonAwsLambdaLayerDependenciesField(PythonFaaSDependencies):
+    required = True
+
+
+class _AWSLambdaBaseTarget(Target):
+    CORE_FIELDS = (
         *COMMON_TARGET_FIELDS,
         OutputPathField,
-        PythonFaaSDependencies,
-        PythonAwsLambdaHandlerField,
         PythonAwsLambdaIncludeRequirements,
+        PythonAwsLambdaIncludeSources,
```

Comment:
I thought it didn't hurt to have a `include_sources=` field on `python_awslambda` functions in addition to layers, so this is in the shared fields. Let me know if you'd prefer it to be specific to layers.

Reply:
It is a bit weird not to, I think? What would it mean to have a function without sources?

Reply:
I guess one could have sources in a layer and only (a single?) third party dep in a layer, and it uses env vars to call into the sources... but that's pretty weird. I've moved `include_sources` to just `python_aws_lambda_layer` to reduce the permutations/simplify the docs.